graphql/tests/query.rs

```diff
@@ -174,6 +185,34 @@ fn test_schema(id: DeploymentHash, id_type: IdType) -> Schema {
     type Publisher {
         id: Bytes!
     }
+
+    interface Review {
+        id: ID!
+        body: String!
+        author: User!
+    }
+
+    type SongReview implements Review @entity {
+        id: ID!
+        body: String!
+        song: Song
+        author: User!
+    }
+
+    type BandReview implements Review @entity {
+        id: ID!
+        body: String!
+        band: Band
+        author: User!
+    }
+
+    type User @entity {
+        id: ID!
+        name: String!
+        bandReviews: [BandReview] @derivedFrom(field: \"author\")
+        songReviews: [SongReview] @derivedFrom(field: \"author\")
+        reviews: [Review] @derivedFrom(field: \"author\")
+    }
```

Comment:
It looks like all the interface fields in these tests are `@derivedFrom`. Could some of them be made into direct fields? In general, we want to make sure that all four combinations of derived/direct and single value/array fields work.

Reply:
I added a few new test scenarios but I don't know exactly how the interfaces are supposed to work, what is possible, and what is not. I'm guessing here a bit. One test is failing but it yells at me about some non-existing operator (`operator does not exist: bytea = text`). I'm lost at this point :)

Reply:
That error means that somewhere where a SQL query compares two columns with `=`, one is of type `bytea` and the other of type `text`. That's an error somewhere in query generation, not necessarily connected to child filters. I wanted to try that out myself, but couldn't build this branch since `cargo` gets unhappy when it tries to check out `https://github.com/dotansimha/graphql-tools-rs?branch=clone-validation-error#89085f31` and says
```
object not found - no match for id (89085f316a71b390fecd9bd4c3f55c3884c33169); class=Odb (9); code=NotFound (-3)
```

The best way to get to the bottom of this is to turn on logging of all statements in Postgres with `alter system set log_statement = 'all';` in `psql` in your test database. (You might have to restart Postgres after this) With this set, run the failing test and the Postgres logs will have the query that is causing that error. Actually, I just noticed that the error message from the test has the full query, though I still recommend setting this.